### PR TITLE
Remove unnecessary vertical scrollbar from inside table widget

### LINKED ISSUE #2895: [Bug] Table has a vertical scrollbar even if content fits
## Description

A table always has a vertical scrollbar, whether or not the content of the table fits within the page. 

### Expected Behavior

The table should never need a vertical scrollbar, because it should load just as many rows as it can display on a single page. 

### Steps to reproduce the behaviour:

1. Open any table
2. Try scrolling in the table

Here's a loom recording showing what I mean:
[LOOM DEMO](https://www.loom.com/share/9c895fb2a2124d479ae4f72cafb444f6)

### Important Details
 - Version: Self-Hosted v1.3.3
 - OS: MacOSX (client), Ubuntu (server)
 - Browser: Chrome, Firefox (probably others too)

### PR DESCRIPTION
**Pull request by vicky-primathon**
## Description
Remove unnecessary scrollbar from inside table widget

Fixes #2895 #3249 

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by riodeuno** [comment]
@vicky-primathon Let's make the scrollbar a virtual scrollbar, which will not take up space in the table itself.

**Comment by riodeuno** [comment]
@vicky-primathon 
- Horizontal scrollbar never shows, it should always show for users with mouse
- Vertical scrollbar should show when scrolling
- Header scrollbar should also be virtual scrollbar

File: app/client/src/components/ads/HorizontalScrollIndicator.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+const ScrollTrack = styled.div<{
+  isVisible: boolean;
+  bottom?: string;
+  left?: string;
+  mode?: "DARK" | "LIGHT";
+}>`
+  position: absolute;
+  z-index: 100;
+  bottom: ${(props) => (props.bottom ? props.bottom : "0px")};
+  left: ${(props) => (props.left ? props.left : "0")};
```
Comment (riodeuno) [inline app/client/src/components/ads/HorizontalScrollIndicator.tsx:15]: Let's move the common styles to `DefaultTheme.tsx`

File: app/client/src/components/ads/HorizontalScrollIndicator.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+  }, [isScrollVisible]);
+
+  const hideScrollbar = _.debounce(() => {
+    setIsScrollVisible(alwaysShowScrollbar || false);
+  }, 1500);
+
+  return (
+    <ScrollTrack isVisible={isScrollVisible} bottom={bottom} left={left}>
+      <ScrollThumb
+        className="scrollbar-thumb"
```
Comment (riodeuno) [inline app/client/src/components/ads/HorizontalScrollIndicator.tsx:114]: Let's remove this className

File: app/client/src/components/designSystems/appsmith/TableComponent/Table.tsx [hunk 3]
```diff
@@ -66,7 +67,7 @@ const defaultColumn = {
 
 export const Table = (props: TableProps) => {
   const isResizingColumn = React.useRef(false);
-  const tableWrapperRef = React.useRef<HTMLDivElement>(null);
+  // const tableWrapperRef = React.useRef<HTMLDivElement>(null);
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/TableComponent/Table.tsx:70]: Let's remove commented code.

File: app/client/src/components/designSystems/appsmith/TableComponent/Table.tsx [hunk 4]
```diff
@@ -269,9 +275,20 @@ export const Table = (props: TableProps) => {
                 prepareRow,
               )}
           </div>
+          <HorizontalScrollIndicator
+            containerRef={tableWrapperRef}
+            mode="LIGHT"
+            alwaysShowScrollbar
+          />
         </div>
+        <ScrollIndicator
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/TableComponent/Table.tsx:284]: Let's rename this component to be `VerticalScrollIndicator`

File: app/client/src/components/designSystems/appsmith/TableComponent/TableStyledWrappers.tsx [hunk 5]
```diff
@@ -357,14 +384,22 @@ export const TableHeaderWrapper = styled.div<{
...
   display: flex;
   border-bottom: 1px solid ${Colors.GEYSER_LIGHT};
-  width: ${(props) => props.width}px;
+  width: ${(props) => props.width - 8}px;
   .show-page-items {
     display: ${(props) => (props.width < 700 ? "none" : "flex")};
   }
   overflow-x: auto;
   overflow-y: hidden;
+  scrollbar-width: none;
```
Comment (riodeuno) [inline app/client/src/components/designSystems/appsmith/TableComponent/TableStyledWrappers.tsx:396]: Let's create a css called `DisableNativeScrollbar` and use that.

**Review by riodeuno — CHANGES_REQUESTED** [review]

**Comment by riodeuno** [comment]
@vicky-primathon The horizontal scrollbar isn't showing. I'm on Mac, Chrome, external mouse and keyboard.

File: app/client/src/pages/Editor/PropertyPane/index.tsx [hunk 6]
```diff
@@ -32,7 +32,7 @@ import { deleteSelectedWidget, copyWidget } from "actions/widgetActions";
 import { ControlIcons } from "icons/ControlIcons";
 import { FormIcons } from "icons/FormIcons";
 import PropertyPaneHelpButton from "pages/Editor/PropertyPaneHelpButton";
-// import ScrollIndicator from "components/ads/ScrollIndicator";
+// import VerticalScrollIndicator from "components/ads/VerticalScrollIndicator";
```
Comment (riodeuno) [inline app/client/src/pages/Editor/PropertyPane/index.tsx:35]: Let's remove this commented code.

**Comment by areyabhishek** [comment]
@vicky-primathon @somangshu When can this PR be closed?

**Comment by somangshu** [comment]
> @vicky-primathon @somangshu When can this PR be closed?

@areyabhishek we are now testing the deploy preview and should get merged max by mid next week.

**Comment by somangshu** [comment]
#4006 is created on top of this.

**Comment by mohanarpit** [comment]
@somangshu @vicky-primathon Can we close this in favour of #4006 then?

**Comment by riodeuno** [comment]
Closing in favour of #4006

**Pull request closed without merging**